x-pack/plugins/advanced_ui_actions/public/services/ui_actions_service_enhancements.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ActionFactoryRegistry } from '../types';
+import { ActionFactory, ActionFactoryDefinition } from '../dynamic_actions';
+
+export interface UiActionsServiceEnhancementsParams {
+  readonly actionFactories?: ActionFactoryRegistry;
+}
+
+export class UiActionsServiceEnhancements {
+  protected readonly actionFactories: ActionFactoryRegistry;
+
+  constructor({ actionFactories = new Map() }: UiActionsServiceEnhancementsParams = {}) {
+    this.actionFactories = actionFactories;
+  }
+
+  /**
+   * Register an action factory. Action factories are used to configure and
+   * serialize/deserialize dynamic actions.
+   */
+  public readonly registerActionFactory = <
+    Config extends object = object,
+    FactoryContext extends object = object,
```

Comment:
Should these two types extend the `ActionBaseConfig` and `ActionFactoryBaseContext` type instead of `object`?

Reply:
No, I've removed those two interface, they are not needed.